File: sample_code/operatorsdemo.go
package main

/**	This program program has no output.
  *	It demonstrates most of the operators I know of.
  *	If you know of any other operators, please add them.
  */

func main() {
	// arithmetic operators
    var1 := 1;
    var2 := 2;
	arr := []int{ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };

	var3 := var1 + var2;
	var3 = var1 - var2;
	var3 = var1 * var2;
	var3 = var1 / var2;
	var3 = var1 % var2;

	var3 = +var1;
	var3 = -var1;
	var1++;
	var1--;

	// relational operators
	eq := var1 == var2;
	eq = var1 != var2;
	eq = var1 < var2;
	eq = var1 > var2;
	eq = var1 <= var2;
	eq = var1 >= var2;

	// logical operators
	eq = !eq;
	eq = eq && true;
	eq = eq || false;

	// bitwise operators
	var3 = ^var1;
	var3 = var1 & var2;
	var3 = var1 | var2;
	var3 = var1 ^ var2;
	var3 = var1 &^ var2;
	var3 = var1 >> 1;
	var3 = var1 << 1;

	// compound assignment operators
	var3 += var2;
	var3 -= var2;
	var3 *= var2;
	var3 /= var2;
	var3 %= var2;
	var3 &= var2;
	var3 |= var2;
	var3 ^= var2;
	var3 >>= 1;
	var3 <<= 1;

	// other
	var3 = arr[ 2 ];
}

File: sample_code/operatorsdemo.py
"""
This program has no output.
It demonstrates most of the operators I know of.
If you know of any other operators, please add them.
"""

def main():

	# arithmetic operators
	var1 = 1;
	var2 = 2;
	var3 = var1;
	arr = [ i for i in range( 10 ) ]

	var1 + var2;
	var1 - var2;
	var1 * var2;
	var1 ** var2;
	var1 / var2;
	var1 // var2;
	var1 % var2;

	"test" + " string"

	+var1;
	-var1;

	# relational operators
	eq = var1 == var2;
	eq = var1 != var2;
	eq = var1 > var2;
	eq = var1 < var2;
	eq = var1 >= var2;
	eq = var1 <= var2;
	eq = var1 is var2;
	eq = var1 is not var2;

	1 in arr
	11 not in arr

	# logical operators
	not eq;
	eq and True;
	eq or False;

	# bitwise operators
	~var1;
	var1 & var2;
	var1 | var2;
	var1 ^ var2;
	var1 >> var2;
	var1 << var2;

	# compound assignment operators
	var1 += var2;
	var1 -= var2;
	var1 *= var2;
	var3 /= var2;
	var1 %= var2;
	var1 **= var2;
	var1 //= var2;
	var1 &= var2;
	var1 |= var2;
	var1 ^= var2;
	var1 >>= var2;
	var1 <<= var2;

	# other
	arr[ 2 ]
	arr[ 1 : 5 ]

if __name__ == "__main__":
	main()
